Source/Core/DiscIO/FileSystemGCWii.cpp

```diff
@@ -270,6 +271,19 @@ void CFileSystemGCWii::InitFileSystem()
 	if (!Root.IsDirectory())
 		return;
 
+	static const u32 ARBITRARY_FILE_SYSTEM_SIZE_LIMIT = 2 * 1024 * 1024;
+	if (Root.m_FileSize > ARBITRARY_FILE_SYSTEM_SIZE_LIMIT)
```

Comment:
Unfortunately, GetSize() isn't always correct. Some compressed formats return the space the disc image takes after compression, and I think VolumeDirectory always returns 0.